src/components/themed/WalletList.tsx

```diff
@@ -78,22 +79,23 @@ export function WalletList(props: Props): React.ReactElement {
   )
   const sortedWalletList = useSelector(state => state.sortedWalletList)
 
-  const handlePress = React.useMemo(
-    () =>
-      onPress ??
-      ((walletId: string, tokenId: EdgeTokenId): void => {
-        dispatch(selectWalletToken({ navigation, walletId, tokenId })).catch(
-          (error: unknown) => {
-            showError(error)
-          }
-        )
-      }),
-    [dispatch, navigation, onPress]
+  const handlePress = useHandler(
+    async (walletId: string, tokenId: EdgeTokenId) => {
+      if (onPress != null) {
+        await onPress(walletId, tokenId)
+      } else {
+        await dispatch(
+          selectWalletToken({ navigation, walletId, tokenId })
+        ).catch((error: unknown) => {
+          showError(error)
+        })
```

Comment:
We don't actually need this catch block, because the `WalletListCurrencyRow` handles errors via `EdgeTouchableOpacity`.